modules/core/src/v2/coins/eth.ts

```diff
@@ -170,8 +170,14 @@ export interface RecoverOptions {
   walletContractAddress: string;
   recoveryDestination: string;
   krsProvider?: string;
+  gasPrice?: number;
+  gasLimit?: number;
 }
 
+export interface GasInfo {
+  gasPrice: any;
+  gasLimit: any;
```

Comment:
I would have made the types "BN", which is a variation of BigNumber.  However, `BN` is a type in the `ethereum-util` library, which is only imported dynamically upon use. Therefore, I can't really get Typescript type to work with it.

Reply:
If the types are part of our API, we should depend on the library (or at least the types package) ourselves so we can safely export the type.

Reply:
How come these are `any` here when they are `number` in other places? Also, is there any place where these need to be numeric strings?

Reply:
> How come these are any here when they are number in other places? Also, is there any place where these need to be numeric strings?

What you see on line 173 and 174 are the input formats, which are numbers. But `GasInfo` is the output type of `checkGasPriceAndLimit`, which were expected to be BN in the original implementation. So I made them BN to work with the old code.